green-hub/src/components/Ideas/Ideas.jsx

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect, useState } from "react";
+import { Navbar, Footer } from "../index";
+import { db } from "../../config/firebase.js";
+import {
+  Timestamp,
+  addDoc,
+  getDocs,
+  collection,
+  orderBy,
+  onSnapshot,
+  query,
+} from "firebase/firestore";
+import { Toaster, toast } from "react-hot-toast";
+import IdeaCard from "../IdeaCard/IdeaCard";
+import styles from "./Ideas.module.css";
+import useAuth from "../../context/AuthContext";
+
+const Ideas = () => {
+  const [idea, setIdea] = useState([]);
+  const [user, setUser] = useState('')
+  const ideasCollectionRef = collection(db, "ideas");
+  const usersCollectionRef = collection(db, "users")
+  const { currentUser } = useAuth();
+
+  // POBIERANIE USERÓW
+  const getUser = async () => {
+    try {
+      const data = await getDocs(usersCollectionRef);
+      const filteredData = data.docs.map((doc) => ({
+        ...doc.data()
+      }))
+      const user = filteredData.filter(user => user.email === currentUser.email)
+      const userName = `${user[0].name} ${user[0].lastName}`
+      console.log(userName)
+      setUser(userName);
+    } catch {
+      console.log("no user here");
+    }
+  }
+  
+  useEffect(() => {
+    getUser();
+  }, [])
+
+  // WYŚWIETLANIE POMYSŁÓW UŻYTKOWNIKÓW
+  const q = query(ideasCollectionRef, orderBy("date", "desc"));
+
+  const getIdeasFromSnapshot = (querySnapshot) => {
+    return querySnapshot.docs.map((doc) => ({
+      id: doc.id,
+      ...doc.data(),
+    }));
+  };
+
+  const dbListener = (cb) => onSnapshot(q, cb);
+
+  useEffect(() => {
+    dbListener((querySnapshot) => setIdea(getIdeasFromSnapshot(querySnapshot)));
+  }, []);
+
+  // funkcja do stworzenia obiektu z nowym pomysłem przesłanym przez użytkownika
+  const getNewIdea = (e) => {
+    const newIdea = {
+      user: user,
+      idea: e.target.idea.value,
+      date: Timestamp.fromDate(new Date()),
+      auth: currentUser.uid
```

Comment:
czyli pomysł jest taki, żeby do pomysłu wrzucać ID użytkownika i potem to sprawdzać, zamiast wrzucać pomysły do Usera, dobrze rozumiem? Jeśli tak, to ok, można zignorować powyższą uwagę o tym, że do Usera też powinna wlecieć informacja o pomysłach :)